### PR TITLE
feat: part 1

### PR DESCRIPTION
**Pull request by bh90210**
- [x] memset
- [x] bzero
- [x] memcpy
- [x] memccpy
- [x] memmove
- [x] memchr
- [x] memcmp
- [x] strlen
- [x] strlcpy
- [x] strlcat
- [x] strchr
- [x] strrchr
- [x] strnstr
- [x] strncmp
- [x] atoi
- [x] isalpha
- [x] isdigit
- [x] isalnum
- [x] isascii
- [x] isprint
- [x] toupper
- [x] tolower
- [x] calloc
- [x] strdup

### CODE REVIEW

**Review by bh90210 — COMMENTED** [review]
# needs fixing
- [x] atoi
- [x] strncmp
- [x] strrchr
- [x] strchr
- [x] strrchr
- [x] strlcat

**Review by bh90210 — COMMENTED** [review]
most should be `OK`, there is one problem with `memmove` but I am on it :+1:

**Review by bh90210 — COMMENTED** [review]
all done, all tests passing, please put some human input :)

File: ft_memmove.c [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+			*(char *) dest++ = *(char *) src++;
+	else
+	{
+		src += n;
+		dest += n;
+		while (n--)
+			*(char *)--dest = *(char *)--src;
+	}
+	return (d);
+}
```
Comment (bh90210) [inline ft_memmove.c:31]: https://stackoverflow.com/questions/13339582/why-is-linux-memmove-implemented-the-way-it-is

**Review by ignan1 — CHANGES_REQUESTED** [review]
ft_strnstr is failing when s2 is only 1 character long

**Review by bh90210 — COMMENTED** [review]
fixed :+1:

**Pull request merged**